### ISSUE TITLE
Bad error description on invalid email link in paradoxValidateLinks

### ISSUE THREAD

**Issue by Dollyg** [lightbend/paradox#411]
I have this text in my docs `x@y.z` which is not a valid email link.

The error I get is very undescriptive. 
```
(docs / Paradox / paradoxValidateLinks) java.lang.IllegalArgumentException: Illegal character in fragment at index 15: mailto:&#120;@&#121;&#x2e;&#x7a;
``` 

Having atleast filename, linenumber would be very helpful.

**Comment by raboof** [comment]
I agree! Would you be interested in trying to improve this? I think an `ErrorContext` is available during most of the validation, if you can catch this IllegalArgumentException and report it through `ErrorContext`, including the file that is being validated, that should help it report a better error.

**Comment by Dollyg** [comment]
Will definitely give it a try. Thanks!
